src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1172 @@ const ActorInit En_Kakasi_InitVars = {
+    ENKAKASI_ANIME_SIDEWAYS_SHAKING = 1,
+    ENKAKASI_ANIME_HOPPING_REGULAR = 2,    // bounce bounce bounce (repeat)
+    ENKAKASI_ANIME_SPIN_REACH_OFFER = 3,   // spin and then reaching toward player
+    ENKAKASI_ANIME_TWIRL = 4,              // slower cinimatic twirl
+    ENKAKASI_ANIME_ARMS_CROSSED_STILL = 5, // arms crossed but still, also some noise sfx
+    ENKAKASI_ANIME_WAVE = 6,               // "wave" short sideways shake, stops early, partial? unused?
+    ENKAKASI_ANIME_SLOWROLL = 7,           // partial bounch, ends looking left, OFFER anim takes over
+    ENKAKASI_ANIME_IDLE = 8,               // slow stretching wiggle, ends in regular position
+
+} EnKakasi_Animations;
+
+extern AnimationHeader D_06007444;
+extern AnimationHeader D_0600686C;
+extern AnimationHeader D_060081A4;
+extern AnimationHeader D_06007B90;
+extern AnimationHeader D_060071EC;
+extern AnimationHeader D_06007444;
+extern AnimationHeader D_0600686C;
+extern AnimationHeader D_060081A4;
+extern AnimationHeader D_06000214;
+
+// static AnimationHeaderCommon* kakasiAnimations[] = {
+static AnimationHeader* kakasiAnimations[] = {
+    &D_06007444, &D_0600686C, &D_060081A4, &D_06007B90, &D_060071EC, &D_06007444, &D_0600686C, &D_060081A4, &D_06000214,
+};
+
+// animation mode, passed to SkelAnime_ChangeAnim
+u8 D_8097206C[] = {
+    0x00, 0x00, 0x00, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+};
+
+void EnKakasi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi* this = (EnKakasi*)thisx;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+extern FlexSkeletonHeader D_060065B0;
+
+void EnKakasi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi* this = (EnKakasi*)thisx;
+    s32 tempCutscene;
+    s32 i;
+
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &D_80971D80);
+    SkelAnime_InitSV(globalCtx, &this->skelanime, &D_060065B0, &D_06000214, 0, 0, 0);
+
+    this->songSummonDist = GET_KAKASI_SUMMON_DISTANCE(this) * 20.0f;
+    if (this->songSummonDist < 40.0f) {
+        this->songSummonDist = 40.0f;
+    }
+
+    this->unkHeight = (this->actor.world.rot.z * 20.0f) + 60.0f;
+    this->actor.world.rot.z = 0;
+    this->actor.targetMode = 0;
+    if (this->actor.world.rot.x > 0 && this->actor.world.rot.x < 8) {
+        this->actor.targetMode = this->actor.world.rot.x - 1;
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+
+    this->aboveGroundStatus = GET_KAKASI_ABOVE_GROUND(this);
+    this->actor.world.rot.x = 0;
+    this->actor.flags |= 0x400;
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    Actor_SetScale(&this->actor, 0.01f);
+
+    i = 0;
+    tempCutscene = this->actor.cutscene;
+    while (tempCutscene != -1) {
+        // petrie911 has suggested this might be a fake match, no others found yet
+        tempCutscene = ActorCutscene_GetAdditionalCutscene(this->actorCutscenes[i] = tempCutscene);
+        i++;
```

Comment:
```suggestion
    while (tempCutscene != -1) {
        // This might be a fake match, no others found yet
        tempCutscene = ActorCutscene_GetAdditionalCutscene(this->actorCutscenes[i] = tempCutscene);
        i++;
```